android/src/main/kotlin/com/airship/flutter/ConfigManager.kt

```diff
@@ -26,49 +31,246 @@ class ConfigManager(private val context: Context) {
         }
     }
 
-    private val Context.airshipFlutterPluginStore: DataStore<Preferences> by preferencesDataStore(
-        name = "airshipFlutterPlugin"
+
+    private val Context.flutterPluginStore: DataStore<Config.AirshipConfig> by dataStore(
+        fileName = "airship.flutter.plugin.pb",
+        ConfigSerializer
     )
 
-    suspend fun updateConfig(appKey: String, appSecret: String) {
-        context.airshipFlutterPluginStore.edit { preferences ->
-            preferences[APP_KEY] = appKey
-            preferences[APP_SECRET] = appSecret
+    suspend fun updateConfig(configByteArray: ByteArray) {
+        context.flutterPluginStore.updateData {
+            ConfigSerializer.readFrom(  configByteArray.inputStream())
         }
     }
 
     val config: Flow<AirshipConfigOptions>
         get() = flow {
-            val appCredentialsOverride =
-                context.airshipFlutterPluginStore.data.map { preferences ->
-                    Pair(preferences[APP_KEY], preferences[APP_SECRET])
-                }.first()
+            /// Plugin config stored/ from dart config
+            val pluginConfig = context.flutterPluginStore.data.first()
 
             val config = AirshipConfigOptions.newBuilder()
                 .applyDefaultProperties(context)
+                .setRequireInitialRemoteConfigEnabled(pluginConfig.requireInitialRemoteConfigEnabled)
                 .apply {
-                    if (!(appCredentialsOverride.first.isNullOrEmpty() || appCredentialsOverride.second.isNullOrEmpty())) {
-                        this.setAppKey(appCredentialsOverride.first)
-                            .setAppSecret(appCredentialsOverride.second)
-                            .build()
+                    pluginConfig.inProduction.let {
+                        this.setInProduction(it)
+                    }
+
+                    pluginConfig.development.let {
+                        if (it.isNotEmptyOrPartial) {
+                            this.setDevelopmentAppKey(it.appKey)
+                                .setDevelopmentAppSecret(it.appSecret)
+                                .setDevelopmentLogLevel(it.logLevel.value)
+                                .build()
```

Comment:
`build()` doesn't need to be called here and is throwing warnings into the logs about the allow lists not being set on the config